compat/operator/startWith.ts

```diff
@@ -26,5 +26,5 @@ export function startWith<T>(this: Observable<T>, ...array: Array<T | SchedulerL
  * @owner Observable
  */
 export function startWith<T>(this: Observable<T>, ...array: Array<T | SchedulerLike>): Observable<T> {
-  return higherOrder(...array)(this);
+  return higherOrder<T, T>(...array)(this);
```

Comment:
Is this OK 😅 ?

Reply:
I seem to always forget the `compat` signatures. They should be updated, too.

I need to make myself a checklist of things to do upon making a typings change.

Reply:
Done, thanks for your help!